### ISSUE TITLE
Cannot recover after disconnected from MSSQL server

### ISSUE THREAD

**Issue by GPHemsley** [HeidiSQL/HeidiSQL#1094]
## Steps to reproduce this issue

1. Connect to MSSQL server.
2. Get disconnected.

## Current behavior

* All data views are updated so that values are shown as "Node".
* Attempt to reconnect throw an error about not being able to reopen a closed object.

## Expected behavior

* Connection is re-established automatically or at least gracefully.
* No major error is shown.

## Environment

* **HeidiSQL version:** Version 11.0.0.6049 (64 Bit)
* **Database system and version:**  MSSQL 13.0 (SQL Server 2016)
* **Operating system:** Windows 10

**Comment by ansgarbecker** [comment]
This is similar to #904

After that "Cannot reopen a closed object" error, does HeidiSQL then connect?

**Comment by GPHemsley** [comment]
No, not as far as I can tell. My solution when this happens is always to close HeidiSQL and restart it.

**Comment by GPHemsley** [comment]
This issue may only apply if the connection is terminated client-side (e.g. by losing Internet connection).

I just got the following error, and HeidiSQL re-established the connection successfully:
```
/* SQL Error (10054): TCP Provider: An existing connection was forcibly closed by the remote host
Communication link failure */
```

**Comment by GPHemsley** [comment]
May be related to #1011.

**Comment by GPHemsley** [comment]
There is another error message which I presume is related:
> /* SQL Error (0): Operation is not allowed when the object is closed
unknown */

In this case, once a connection to a MSSQL server is severed, it cannot be reconnected by hitting "refresh", as it will always fail with the above error. I have confirmed that MySQL does not have this problem - refresh works there.
